src/sfdc46/salesforceBulk/csv/csv_delete_operator.bal

```diff
@@ -123,35 +120,9 @@ public type CsvDeleteOperator client object {
     # + batchId - batch ID
     # + numberOfTries - number of times checking the batch state
     # + waitTime - time between two tries in ms
-    # + return - Batch result as CSV if successful else SalesforceError occured
-    public remote function getBatchResults(string batchId, int numberOfTries = 1, int waitTime = 3000) 
+    # + return - Results array if successful else SalesforceError occured
+    public remote function getResult(string batchId, int numberOfTries = 1, int waitTime = 3000) 
         returns @tainted Result[]|SalesforceError {
-        int counter = 0;
-        while (counter < numberOfTries) {
-            Batch|SalesforceError batch = self->getBatchInfo(batchId);
-            
-            if (batch is Batch) {
-
-                if (batch.state == COMPLETED) {
-                    string|SalesforceError result = 
-                        self.httpBaseClient->getCsvRecord([JOB, self.job.id, BATCH, batchId, RESULT]);
-                    if (result is string) {
-                        return getBatchResults(result);
-                    } else {
-                        return result;
-                    }
-                } else if (batch.state == FAILED) {
-                    return getFailedBatchError(batch);
-                } else {
-                    printWaitingMessage(batch);
-                }
-
-            } else {
-                return batch;
-            }
-            runtime:sleep(waitTime); // Sleep 3s.
-            counter = counter + 1;
-        }
-        return getResultTimeoutError(batchId, numberOfTries, waitTime);
+        return checkBatchStateAndGetResults(getBatchPointer, getResultsPointer, self, batchId, numberOfTries, waitTime);
```

Comment:
Would it be better if we can rename these function pointer names?

Reply:
Added an issue to fix this https://github.com/wso2-ballerina/module-salesforce/issues/67